memo_app.rb

```diff
@@ -13,54 +43,29 @@
 end
 
 post '/memos' do
-  max_id = 0
-  memos = JSON.parse(File.read('data.json'))
-  memos['memos'].each do |memo|
-    max_id = memo['id'].to_i if max_id < memo['id'].to_i
-  end
-  new_data = { "id": (max_id + 1).to_s, "title": params[:title], "body": params[:body] }
-  memos['memos'].push(new_data)
-  File.write('data.json', memos.to_json)
+  Memo.create_memo(connection, params[:title], params[:body])
   redirect '/'
 end
 
 get '/memos/:id' do
-  @title = ''
-  @body = ''
-  find_data = find_memo(params[:id])
+  find_data = Memo.find_memos(connection, params[:id])
   @memo = find_data unless find_data.empty?
```

Comment:
```suggestion
  @memo = Memo.find_memos(connection, params[:id])
```

これでよい（やってることは実質変わらない）と思います。